data/src/main/java/com/example/data/network/RequestTokenApi.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.data.network
+
+import com.example.data.common.Constants.CONSUMER_KEY
+import com.example.data.common.Constants.CONSUMER_SECRET
+import com.example.data.common.Constants.URL_ACCESS_TOKEN
+import com.example.data.common.Constants.URL_AUTHORIZE
+import com.example.data.common.Constants.URL_REQUEST_TOKEN
+import kotlinx.coroutines.*
+import oauth.signpost.OAuth
+import oauth.signpost.OAuthConsumer
+import oauth.signpost.OAuthProvider
+import oauth.signpost.commonshttp.CommonsHttpOAuthConsumer
+import oauth.signpost.commonshttp.CommonsHttpOAuthProvider
+
+class RequestTokenApi {
+
+    fun  httpRequestToken()= GlobalScope.async{
```

Comment:
GlobalScope is nor recommended to use. And actually we don't need it.
We need only suspend function here. You can create it using withContext.